# Module 2 Lab 1: Python Warm-Up + First Push

**Name:** Azra

This notebook works through Steps 1–6 of the Module 2 Lab 1 assignment: core Python constructs, a dictionary, and loading/inspecting `chickwts.csv` with pandas.

## Step 1: Wages with f-strings

In [1]:
name = "Azra"
hourly_wage = 32.00
hours = 20
weekly_pay = hourly_wage * hours
print(f"{name} earns {weekly_pay} per week")

Azra earns 640.0 per week


## Step 2: Count the warm days

In [2]:
temps = [61, 65, 72, 58, 70, 74, 68]
warm_days = 0
for t in temps:
    if t > 65:
        warm_days = warm_days + 1
print(warm_days)

4


## Step 3: Write a temperature converter

In [3]:
def to_celsius(f):
    return (f - 32) * 5 / 9

print(round(to_celsius(temps[0]), 1))

16.1


**Checkpoint:** Steps 1–3 ran top to bottom with no `NameError`, and the warm-day count matches the expected value of 4.

## Step 4: A movie dictionary

In [4]:
movie = {"title": "Inception", "year": 2010, "rating": 8.8}

for key, value in movie.items():
    print(key, ":", value)

title : Inception
year : 2010
rating : 8.8


## Step 5: Load and inspect chickwts.csv

In [5]:
import pandas as pd

df = pd.read_csv("chickwts.csv")
df.head()

,weight,feed
0,179,horsebean
1,160,horsebean
2,136,horsebean
3,227,horsebean
4,217,horsebean


In [6]:
df.shape

(71, 2)

In [7]:
df.describe()

,weight
count,71.000000
mean,261.309859
std,78.073700
min,108.000000
25%,204.500000
50%,258.000000
75%,323.500000
max,423.000000


## Step 6: Interpret describe()

The `describe()` output shows 71 chicks with a mean weight of about 261 grams and a median (50th percentile) close to that, so the distribution isn't badly skewed. The spread is wide, though: weights range from about 108 grams to 423 grams, with a standard deviation near 78 grams, meaning individual chicks vary a lot around the average. That's not surprising once you remember `weight` is pooled across six different feed types — the variation likely reflects real differences in how effective each feed was, not measurement noise, which is exactly what you'd want to check next by grouping `weight` by `feed`.

## Appendix: AI Use Disclosure

**Tool used:** Claude (Anthropic), via Claude Code / Cowork.

**How it was used:** Drafted the code cells for Steps 1–5 following the patterns given in the lab's tutorials, and produced a first-draft interpretation for Step 6 that I reviewed and confirmed reflects the actual `describe()` output for this dataset.

**Key prompts:**
- "Build the Module 2 Lab 1 notebook (steps 1–6) following the tutorial patterns exactly, using chickwts.csv."
- "Write a Step 6 interpretation of describe() for the chick weights."